cli/azd/pkg/apphost/generate_types.go

```diff
@@ -93,8 +93,8 @@ type genDaprComponent struct {
 }
 
 type genInput struct {
-	Secret           bool
-	DefaultMinLength int
+	Secret  bool
+	Default interface{}
```

Comment:
Was this done to avoid `apphost` dependency on `provisioning`? I do feel that any external contracts should be called out explicitly. Personally, I would not mind if we just created the structs with the same shape and members in both places. It may give us the flexibility needed if the contracts in two very different features ever diverged.

Reply:
> Was this done to avoid apphost dependency on provisioning?

Not really. I was creating one struct initially, but I realized we don't need to unmarshal the bytes into one structure, as azd doesn't care about any of the contents of the input during infrastructure generation. There are no decisions or computation required for the configuration to auto-generate an input.  azd just unmarshalls it to bytes and marshall it back as a string to the bicep file.

Only the package responsible for generating the values is the one who cares about unmarshalling to the expected contract.